src/catalog/src/durable/transaction.rs

```diff
@@ -1912,16 +1982,18 @@ where
     /// Iterates over the items viewable in the current transaction, and provides a
     /// map where additional pending items can be inserted, which will be appended
     /// to current pending items. Does not verify uniqueness.
-    fn for_values_mut<F: FnMut(&mut BTreeMap<K, Option<V>>, &K, &V)>(&mut self, mut f: F) {
+    fn for_values_mut<F: FnMut(&mut BTreeMap<K, Vec<(V, u64, i64)>>, &K, &V)>(&mut self, mut f: F) {
         let mut pending = BTreeMap::new();
         self.for_values(|k, v| f(&mut pending, k, v));
-        self.pending.extend(pending);
+        for (k, updates) in pending {
+            self.pending.entry(k).or_default().extend(updates);
+        }
     }
 
     /// Inserts a new k,v pair.
     ///
     /// Returns an error if the uniqueness check failed or the key already exists.
-    fn insert(&mut self, k: K, v: V) -> Result<(), DurableCatalogError> {
+    fn insert(&mut self, k: K, v: V, t: u64) -> Result<(), DurableCatalogError> {
```

Comment:
Thoughts on making at least a type alias like `type Timestamp = u64` to make this a tiny bit more explicit? Would help a lot with readability. Diff could get the same treatment.

Reply:
Good idea, just updated.